### ISSUE TITLE
Maven site integration does not work with Maven Site Plugin 3.4

### ISSUE THREAD

**Issue by mosabua** [asciidoctor/asciidoctor-maven-plugin#125]
To reproduce just upgrade the version of the Maven Site Plugin in https://github.com/asciidoctor/asciidoctor-maven-examples/blob/master/asciidoc-site-example/pom.xml
and you will no longer get the html versions of your asciidoc generated..

**Comment by mojavelinux** [comment]
I don't understand how to register multiple file extensions for a site module. If someone knows how to do it, please feel free to file a pull request. It will be most appreciated :)

**Comment by paranoiabla** [comment]
You need to wait for the next release of the doxia plugin. Once that is done you will be able to simply do:
````
    /**
     * The extension for AsciiDoc files.
     */
    public static final String[] FILE_EXTENSION = { "adoc", "asciidoc" };

    /**
     * Build a new instance of {@link AsciidoctorParserModule}.
     */
    public AsciidoctorParserModule()
    {
        super( SOURCE_DIRECTORY, FILE_EXTENSION, AsciidoctorParser.ROLE_HINT );
    }
````

**Comment by mojavelinux** [comment]
Great news!!

**Comment by mosabua** [comment]
@paranoiabla what version of doxia? I might be able to push for a release on the Maven dev list. If its not already in progress..

**Comment by paranoiabla** [comment]
The 2 pullrequests I have linked above (for `maven-doxia` and `maven-doxiasitetools`) need to be merged, a new release of doxia need to be issued. I believe @hboutemy is taking care of that :)

**Comment by mstanleyjones** [comment]
I'm not sure if my problem is relating to this one. Using Maven site plugin 3.4, I can get .adoc files in src/site/asciidoc to build into standalone themed HTML files. However, if a file includes other files, the included files do not get resolved. For instance, in src/site/asciidoc/ I have book.adoc, which includes chapters/installation.adoc, chapters/administration.adoc, etc. If I build book.adoc using an execution of the asciidoctor plugin, it works. However, if I try to build it using mvn site, the includes do not get resolved. Presumably there is a step where everything gets concatenated into one big .adoc before building, and that is not happening with the 'mvn site' target.

I am happy to file this as a separate issue, but it seems that it might be related here so I thought I would post here first.

**Comment by mojavelinux** [comment]
@mstanleyjones The problem here is that the configuration of Asciidoctor within the Maven Site plugin is extremely limited. Likely what is happening is that the includes are not being found because Asciidoctor is being invoked without the proper base directory, and a custom base directory cannot be set. We really need to figure out how we can configure Asciidoctor when it is invoked via the Maven Site plugin. In essence, we need a configuration block like the regular plugin has.

A first step here is to get includes working in the example project so that we know it at least works (and can explore the problem in depth). See https://github.com/asciidoctor/asciidoctor-maven-examples/tree/master/asciidoc-maven-site-example

**Comment by mojavelinux** [comment]
@mstanleyjones As I suspected, the baseDir is not being set and is therefore defaulting to the root of the project. Thus, to resolve an include, it has to begin with `src/site/asciidoc`. That needs to be fixed because it should match the behavior of the main plugin.

**Comment by mojavelinux** [comment]
See #180

**Comment by mojavelinux** [comment]
I believe this issue was resolved in 1.5.2. We're addressing some of the other points raised in this thread as part of 1.5.3.

**Comment by mosabua** [comment]
Awesome news. Looking forward to the new release!
